### PR TITLE
Add loss functions

### PR DESCRIPTION
**Pull request by isaacmg**
The goal of this PR is to add four new losses to flow forecast. QuantileLoss and Guassian will not be usable until a future PR. Goal here is just to get MAPE and RMSE from Kriti working.
- [ ]  Tested on flow 
- [ ] Added unit tests

### CODE REVIEW

File: flood_forecast/model_dict_function.py [hunk 1]
```diff
@@ -18,7 +22,7 @@
 "TransformerXL": TransformerXL, "DummyTorchModel": DummyTorchModel,  "LSTM":LSTMForecast, "SimpleLinearModel": SimpleLinearModel, 
 "CustomTransformerDecoder":CustomTransformerDecoder}
 
-pytorch_criterion_dict = {"MSE": MSELoss(), "SmoothL1Loss":SmoothL1Loss(), "PoissonNLLLoss":PoissonNLLLoss()}
+pytorch_criterion_dict = {"MSE": MSELoss(), "SmoothL1Loss":SmoothL1Loss(), "PoissonNLLLoss": PoissonNLLLoss(), "RMSELoss": RMSE, "MAPE": MAPELoss()}
```
Comment (Pranjalya) [inline flood_forecast/model_dict_function.py:25]: Change RMSE to RMSELoss()

Comment (isaacmg) [reply flood_forecast/model_dict_function.py]: Yeah saw that. Should've waited for unit tests before posting. Thanks.

File: flood_forecast/custom/custom_opt.py [hunk 2]
```diff
@@ -14,27 +16,91 @@ def warmup_cosine(x, warmup=0.002):
...
 
+
 SCHEDULES = {
     'warmup_cosine':   warmup_cosine,
     'warmup_constant': warmup_constant,
     'warmup_linear':   warmup_linear,
 }
 
 
+class RMSELoss(torch.nn.Module):
```
Comment (wwymak) [inline flood_forecast/custom/custom_opt.py:43]: I would suggest putting these loss functions in a separate file called `loss` (i.e. `flood_forecast/custom/loss.py`: `custom_opt` should be all about optimizers...)

File: flood_forecast/custom/custom_opt.py [hunk 3]
```diff
@@ -14,27 +16,91 @@ def warmup_cosine(x, warmup=0.002):
...
+    '''
+    Returns MAPE using:
+    target -> True y
+    output -> Predtion by model
+    '''
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, target: torch.Tensor, output: torch.Tensor):
+        return torch.mean(torch.abs((target - output) / target))
```
Comment (wwymak) [inline flood_forecast/custom/custom_opt.py:68]: do you need to add an epsilon to the denomiator? what happens if target is 0?

File: flood_forecast/custom/custom_opt.py [hunk 4]
```diff
@@ -14,27 +16,91 @@ def warmup_cosine(x, warmup=0.002):
...
+    
+# Add custom loss function
+class GaussianLoss(torch.nn.Module):
+    def __init__(self,mu,sigma):
+        """Compute the negative log likelihood of Gaussian Distribution
+        From https://arxiv.org/abs/1907.00235
+        """
+        super(GaussianLoss, self).__init__()
```
Comment (wwymak) [inline flood_forecast/custom/custom_opt.py:76]: I don't think you need `super(GaussianLoss, self).__init__()` in python 3, although it's still valid. Probably we should just stick with one way of doing it rather than have 2 different ways of calling the super class in the same file...

File: flood_forecast/evaluator.py [hunk 5]
```diff
@@ -84,6 +84,9 @@ def evaluate_model(
             eval_log[target + "_" + evaluation_metric] = s
     return eval_log, df_train_and_test, forecast_start_idx, df_predictions
 
+def infer_on_torch_model(model, test_csv_path:str = None, datetime_start=datetime(2018, 9, 22, 0), hours_to_forecast: int = 336, decoder_params=None, dataset_params:Dict={}, num_prediction_samples:int=None):
```
Comment (wwymak) [inline flood_forecast/evaluator.py:87]: why is there 2 functions with the same name? (isn't this the same as the one on line 91?)

File: flood_forecast/custom/custom_opt.py (hunk 4 above)
Comment (isaacmg) [reply flood_forecast/custom/custom_opt.py:76]: Yeah this was just a copy from the original code of the authors. But I'll fix it

File: flood_forecast/custom/custom_opt.py (hunk 2 above)
Comment (isaacmg) [reply flood_forecast/custom/custom_opt.py:43]: Yeah that is a good point. But I might address that in a future PR.

File: flood_forecast/custom/custom_opt.py (hunk 3 above)
Comment (isaacmg) [reply flood_forecast/custom/custom_opt.py:68]: Good point that will cause division by zero error.

File: flood_forecast/evaluator.py (hunk 5 above)
Comment (isaacmg) [reply flood_forecast/evaluator.py:87]: Yeah good catch that was a remnant of a merge conflict.

File: flood_forecast/custom/custom_opt.py (hunk 3 above)
Comment (isaacmg) [reply flood_forecast/custom/custom_opt.py:73]: But isn't that one of the properties of MAPE that it will cause a division by zero error.
